src/core/ucc_constructor.c

```diff
@@ -37,7 +37,8 @@ static int callback(struct dl_phdr_info *info, size_t size, void *data)
            it'll always write '\0' to the end position of the dest string. */
         ucc_strncpy_safe(component_path, info->dlpi_name, pos + 1);
         component_path[pos] = '\0';
-        strncat(component_path, UCC_COMPONENT_LIBDIR, UCC_COMPONENT_LIBDIR_LEN);
+        strncat(component_path, UCC_COMPONENT_LIBDIR,
```

Comment:
i don't get this. 3rd arg is "how many chars to append" to component_path. 
if component path is :/aaa/bbb/ccc/ddd/eee/fff" then you will try to append here 24-3-1 - 20 ?